src/components/SelectCouponItem/SelectCouponItem.tsx

```diff
@@ -0,0 +1,59 @@
+import Text from '../common/Text/Text';
+import * as S from './SelectCouponItem.style';
+import Checkbox from '../common/Checkbox/Checkbox';
+import { Divider } from '../common/Divider/Divider.style';
+import { toKoreanDate, toKoreanTime } from '../../utils/date';
+import useSelectedCoupon from '../../hooks/coupon/useSelectedCoupon';
+import useSelectedCouponList from '../../hooks/coupon/useSelectedCouponList';
+import { validateCouponAvailable } from '../../utils/validateCouponAvailable';
+
+interface SelectCouponItemProps {
+  coupon: Coupon;
+}
+
+const SelectCouponItem = ({ coupon }: SelectCouponItemProps) => {
+  const { id, description, expirationDate, minimumAmount, availableTime } =
+    coupon;
+
+  const { isSelected, toggleSelectedCoupon } = useSelectedCoupon();
+  const { selectedCouponList } = useSelectedCouponList();
+
+  const disabled =
+    !validateCouponAvailable(coupon) ||
+    (selectedCouponList.length === 2 && !selectedCouponList.includes(coupon));
+
+  if (disabled && isSelected(id)) {
+    toggleSelectedCoupon(coupon);
+  }
+
+  return (
+    <S.SelectCouponItem disabled={disabled}>
```

Comment:
```suggestion
    <S.SelectCouponItem disabled>
```

저도 이번에 알았는데, 요렇게 표현해줄 수도 있더라구요!! 꿀팁 참고! 👍

Reply:
그런데.. 이렇게 되면 disable밖에 못해서
저는 `<S.SelectCouponItem {...{disabled}}>`나 &&연산자를 사용해야하는 걸로 알고있어요..

Reply:
그래서 보통 사용할 때는 false를 기본 값으로 두고 옵셔널로 두면 입력했을 때만 true가 되더라구요!